pyflakes/checker.py

```diff
@@ -552,7 +622,8 @@ class Checker(object):
         builtIns.update(_customBuiltIns.split(','))
     del _customBuiltIns
 
-    def __init__(self, tree, filename='(none)', builtins=None,
+    # TODO: tokens=() is just to make flake8 happy
```

Comment:
yeah this is api shape issue -- I'm a bit stuck here on the right thing to do. Here's what I want and the constraints:

- tree + tokens are required position arguments -- they are necessary to perform all of the checks as implemented in this PR
- flake8 <=3.6.0 calls `Checker` directly with positional arguments (in 3.7.0+ it will call it with tree positionally and `filename=` as a named argument)
- pyflakes uses flake8 to validate itself

I *think* the easiest way forward is to soft-break the api for those which are calling `Checker()` without named arguments and release a minor version with `tokens=()`.  flake8 pins a specific version of pyflakes and can therefore update in lockstep safely.  At a later date, pyfalkes makes `tokens` a required positional argument and releases a major release.

Alternatively, we could just hard-break today and I could change the tests to create separate flake8 environments which don't use the source tree's pyflakes to lint things -- though this is much messier than what I have right now.

I think for option 1, the branch as-is is fine, though I should update the comment

Reply:
yeah I thought about warnings, but there's no precedent in pyflakes currently.  basically I'd do

```python
if not tokens:
    warnings.warn(FutureWarning, 'tokens was not specified, not checking `# type:` comments')
```

though this might be annoying for those that can't control their environments

Reply:
Ah, good point about this ending up showing for end users rather more than the developers of the intermediate packages.
I'm not sure what's best in that case. Perhaps just something in the docs?